src/java.base/share/classes/java/util/FormatItem.java

```diff
@@ -0,0 +1,570 @@
+
+        @Override
+        public abstract long mix(long lengthCoder);
+
+        @Override
+        public abstract long prepend(long lengthCoder, byte[] buffer) throws Throwable;
+    }
+
+    /**
+     * Fill left format item.
+     */
+    static final class FormatItemFillLeft extends FormatItemModifier
+            implements FormatConcatItem {
+        private final int width;
+
+        FormatItemFillLeft(int width, FormatConcatItem item) {
+            super(item);
+            this.width = Integer.max(length(), width);
+        }
+
+        @Override
+        public long mix(long lengthCoder) {
+            return (lengthCoder | coder()) + width;
+        }
+
+        @Override
+        public long prepend(long lengthCoder, byte[] buffer) throws Throwable {
+            MethodHandle putCharMH = selectPutChar(lengthCoder);
+            lengthCoder = item.prepend(lengthCoder, buffer);
+
+            for (int i = length(); i < width; i++) {
+                putCharMH.invokeExact(buffer, (int)--lengthCoder, (int)' ');
+            }
+
+            return lengthCoder;
+        }
+    }
+
+    /**
+     * Fill right format item.
+     */
+    static final class FormatItemFillRight extends FormatItemModifier
+            implements FormatConcatItem {
+        private final int width;
+
+        FormatItemFillRight(int width, FormatConcatItem item) {
+            super(item);
+            this.width = Integer.max(length(), width);
+        }
+
+        @Override
+        public long mix(long lengthCoder) {
+            return (lengthCoder | coder()) + width;
+        }
+
+        @Override
+        public long prepend(long lengthCoder, byte[] buffer) throws Throwable {
+            MethodHandle putCharMH = selectPutChar(lengthCoder);
+
+            for (int i = length(); i < width; i++) {
+                putCharMH.invokeExact(buffer, (int)--lengthCoder, (int)' ');
+            }
+
+            lengthCoder = item.prepend(lengthCoder, buffer);
+
+            return lengthCoder;
+        }
+    }
+
+
+    /**
+     * To upper case format item.
+     */
+    static final class FormatItemUpper extends FormatItemModifier
+            implements FormatConcatItem {
+        FormatItemUpper(FormatConcatItem item) {
+            super(item);
+        }
+
+        @Override
+        public long mix(long lengthCoder) {
+            return (lengthCoder | coder()) + length();
+        }
+
+        @Override
+        public long prepend(long lengthCoder, byte[] buffer) throws Throwable {
+            MethodHandle getCharMH = selectGetChar(lengthCoder);
+            MethodHandle putCharMH = selectPutChar(lengthCoder);
+            lengthCoder = item.prepend(lengthCoder, buffer);
+            int start = (int)lengthCoder;
+
+            for (int i = 0; i < length(); i++) {
+                char ch = (char)getCharMH.invokeExact(buffer, start + i);
+                putCharMH.invokeExact(buffer, start + i, (int)Character.toUpperCase(ch));
+            }
```

Comment:
Some characters may produce multiple uppercase characters, such as "fi" ligature ('\ufb01' -> "FI"), also this should consider locale, e.g. the case for Turkish dotless i.

Reply:
Switch to Formatter if upper selected